[WFLY-17112]: Can't load a custom load balancing policy on a pooled connection factory.

ehsavoie opened the pull request:
* Exporting "org.apache.activemq.artemis.addons" so that the class can be found by Artemis classes.
* Adding a test to cover this.

Jira: https://issues.redhat.com/browse/WFLY-17112

Signed-off-by: Emmanuel Hugonnet <ehugonne@redhat.com>

bstansberry commented on testsuite/integration/manualmode/src/test/java/org/jboss/as/test/manualmode/messaging/ActivemqServerAddonsTestCase.java:
```diff
@@ -0,0 +1,207 @@
+            Files.writeString(addonsModule.resolve("module.xml"), MODULE_XML, StandardOpenOption.CREATE);
+            try ( OutputStream out = Files.newOutputStream(addonsModule.resolve("artemis-lb-policy.jar"), StandardOpenOption.CREATE)) {
+                jar.as(ZipExporter.class).exportTo(out);
+            }
```
Please use org.jboss.as.test.module.util.TestModule.

ehsavoie committed:
[WFLY-17112]: Can't load a custom load balancing policy on a pooled connection factory.

* Exporting "org.apache.activemq.artemis.addons" so that the class can
  be found by Artemis classes.
* Adding a test to cover this.

Jira: https://issues.redhat.com/browse/WFLY-17112

Signed-off-by: Emmanuel Hugonnet <ehugonne@redhat.com>

bstansberry commented on testsuite/integration/manualmode/src/test/java/org/jboss/as/test/manualmode/messaging/ActivemqServerAddonsTestCase.java:
```diff
@@ -0,0 +1,178 @@
+            }
+            final JavaArchive jar = module.addResource("artemis-lb-policy.jar");
+            jar.addClasses(OrderedLoadBalancingPolicy.class);
+            System.setProperty("module.path", Paths.get("target" ,"wildfly", "modules").toAbsolutePath().toString());
```
This is ok but the typical way we deal with this is via module.path in system-property-variables in the pom.

That's not necessarily better, it's just the usual handling that people might look for.

bstansberry reviewed (approved):
I had one comment but that isn't going to stop me merging this.

bstansberry commented:
Thanks @ehsavoie
